tools/src/main/java/org/aya/util/error/SourceFile.java

```diff
@@ -6,23 +6,20 @@
 import org.jetbrains.annotations.NotNull;
 
 import java.nio.file.Path;
-import java.util.Objects;
 
-public record SourceFile(@NotNull Option<Path> path, @NotNull String sourceCode) {
-  public SourceFile(@NotNull Path path, @NotNull String sourceCode) {
-    this(Option.some(path), sourceCode);
+public record SourceFile(
+  @NotNull String display,
+  @NotNull Option<Path> underlying,
+  @NotNull String sourceCode
+) {
+  public SourceFile(@NotNull String display, @NotNull Path underlying, @NotNull String sourceCode) {
+    this(display, Option.some(underlying), sourceCode);
   }
 
-  public static final SourceFile NONE = new SourceFile(Option.none(), "");
-  public static final SourceFile SER = new SourceFile(Option.none(), "");
+  public static final SourceFile NONE = new SourceFile("<unknown-file>", Option.none(), "");
+  public static final SourceFile SER = new SourceFile("<serialized-core>", Option.none(), "");
```

Comment:
Ditto about path names

Reply:
They will not be used to construct a `Path`, so it is safe to keep brackets.

Reply:
Instead, we store the real path in `SourceFile#underlying`

Reply:
Noice